There is a bug somewhere in Cargo/deno/rusty-v8 that causes
build.rs to run twice.  This causes V8 to be build twice at the
same time.  This is an extention of another cludge that stops
this from happening.
diff -ruN v8-0.43.1.orig/build.rs v8-0.43.1/build.rs
--- v8-0.43.1.orig/build.rs	1973-11-29 16:33:09.000000000 -0500
+++ v8-0.43.1/build.rs	2022-06-18 15:50:52.187950075 -0400
@@ -69,7 +69,23 @@
 
   // Build from source
   if env::var_os("V8_FROM_SOURCE").is_some() {
-    return build_v8();
+   // utilize a lockfile to prevent building v8 twice
+    let root = env::current_dir().unwrap();
+    let out_dir = env::var_os("OUT_DIR").unwrap();
+    let lockfilepath = root
+      .join(out_dir)
+      .parent()
+      .unwrap()
+      .parent()
+      .unwrap()
+      .join("building_v8.fslock");
+    println!("building v8 lockfile: {:?}", &lockfilepath);
+    let mut lockfile = LockFile::open(&lockfilepath)
+      .expect("Couldn't open lib download lockfile.");
+    lockfile.lock().expect("Couldn't get lock");
+    build_v8();
+    lockfile.unlock().expect("Couldn't unlock lockfile");
+    return;
   }
 
   // utilize a lockfile to prevent linking of
